src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor_Minimal.cs

```diff
@@ -39,19 +37,7 @@ protected bool NameBoundSuccessfullyToSameSymbol(INamedTypeSymbol symbol)
             ImmutableArray<ISymbol> typeOnlySymbols = semanticModelOpt.LookupNamespacesAndTypes(positionOpt, name: symbol.Name);
             ISymbol typeOnlySymbol = SingleSymbolWithArity(typeOnlySymbols, symbol.Arity);
 
-            if (typeOnlySymbol == null)
-            {
-                return false;
-            }
-
-            var type1 = GetSymbolType(normalSymbol);
-            var type2 = GetSymbolType(typeOnlySymbol);
-
-            return
-                type1 != null &&
-                type2 != null &&
-                type1.Equals(type2) &&
-                typeOnlySymbol.Equals(symbol.OriginalDefinition);
+            return typeOnlySymbol == null ? false : typeOnlySymbol.Equals(symbol.OriginalDefinition);
```

Comment:
We discovered a similar issue with object creation but with simple types rather than type arguments: #21674.